models/_sector/nft/trades/chains/ethereum/platforms/magiceden_ethereum_base_trades.sql

```diff
@@ -0,0 +1,175 @@
+    {% endif %}
+    
+    UNION ALL
+    
+    SELECT evt_block_time AS block_time
+    , evt_block_number AS block_number
+    , evt_tx_hash AS tx_hash
+    , contract_address AS project_contract_address
+    , buyer
+    , seller
+    , tokenAddress AS nft_contract_address
+    , tokenId AS nft_token_id
+    , 1 AS nft_amount
+    , 'Buy' AS trade_category
+    , paymentCoin AS currency_contract
+    , salePrice AS price_raw
+    , evt_index AS sub_tx_trade_id
+    FROM {{ source('limitbreak_ethereum','PaymentProcessor_evt_BuyListingERC1155') }}
+    {% if is_incremental() %}
+    WHERE {{incremental_predicate('evt_block_time')}}
+    {% else %}
+    WHERE evt_block_time >= {{magiceden_start_date}}
+    {% endif %}
+    
+    UNION ALL
+    
+    SELECT evt_block_time AS block_time
+    , evt_block_number AS block_number
+    , evt_tx_hash AS tx_hash
+    , contract_address AS project_contract_address
+    , buyer
+    , seller
+    , tokenAddress AS nft_contract_address
+    , tokenId AS nft_token_id
+    , 1 AS nft_amount
+    , 'Offer Accepted' AS trade_category
+    , paymentCoin AS currency_contract
+    , salePrice AS price_raw
+    , evt_index AS sub_tx_trade_id
+    FROM {{ source('limitbreak_ethereum','PaymentProcessor_evt_AcceptOfferERC721') }}
+    {% if is_incremental() %}
+    WHERE {{incremental_predicate('evt_block_time')}}
+    {% else %}
+    WHERE evt_block_time >= {{magiceden_start_date}}
+    {% endif %}
+    
+    UNION ALL
+    
+    SELECT evt_block_time AS block_time
+    , evt_block_number AS block_number
+    , evt_tx_hash AS tx_hash
+    , contract_address AS project_contract_address
+    , buyer
+    , seller
+    , tokenAddress AS nft_contract_address
+    , tokenId AS nft_token_id
+    , amount AS nft_amount
+    , 'Offer Accepted' AS trade_category
+    , paymentCoin AS currency_contract
+    , salePrice AS price_raw
+    , evt_index AS sub_tx_trade_id
+    FROM {{ source('limitbreak_ethereum','PaymentProcessor_evt_AcceptOfferERC1155') }}
+    {% if is_incremental() %}
+    WHERE {{incremental_predicate('evt_block_time')}}
+    {% else %}
+    WHERE evt_block_time >= {{magiceden_start_date}}
+    {% endif %}
+    )
+    
+, whitelisted_trades AS (
+    SELECT t.block_time
+    , t.block_number
+    , t.tx_hash
+    , t.project_contract_address
+    , t.buyer
+    , t.seller
+    , t.nft_contract_address
+    , t.nft_token_id
+    , t.nft_amount
+    , t.trade_category
+    , t.currency_contract
+    , t.price_raw
+    , t.sub_tx_trade_id
+    , fc.message
+    FROM trades t
+    INNER JOIN {{ source('limitbreak_ethereum','TrustedForwarder_call_forwardCall') }} fc ON fc.call_block_number=t.block_number
+        AND fc.call_tx_hash=t.tx_hash
+        AND fc.contract_address = 0x5ebc127fae83ed5bdd91fc6a5f5767e259df5642
+        {% if is_incremental() %}
+        AND {{incremental_predicate('call_block_time')}}
+        {% else %}
+        AND call_block_time >= {{magiceden_start_date}}
+        {% endif %}
```

Comment:
This is basically only a filter. Is there a reason we don't want to put all of the `PaymentProcessor` events in the model?

Reply:
indeed it is, thats because more entites can use limitbreak's contracts, so this filters to keeep just the ME ones.